### PR TITLE
Rustfmt term, skip formatting statics in compiled

### PR DESCRIPTION
**Pull request by SingingTree**
This is the result of running rustfmt over term. This has previously been done over at libterm in rust, but since I've come to learn that that code mirrors term, I thought it could be of use to have the term formatted. All files have been formatted without any further alterations, except for compiled.rs, which has had its statics skipped, due to the whitespace it results in.

### CODE REVIEW

File: src/terminfo/parm.rs [hunk 1]
```diff
@@ -203,46 +227,55 @@ pub fn expand(cap: &[u8], params: &[Param], vars: &mut Variables)
...
+                            }
+                            None => return Err("stack is empty".to_string()),
+                        }
+                    }
                     'e' => state = SeekIfEnd(0),
                     ';' => (),
                     _ => return Err(format!("unrecognized format option {}", cur)),
                 }
-            },
+            }
```
Comment (Stebalien) [inline src/terminfo/parm.rs]: Any way to put these commas back? This feels inconsistent.

File: src/terminfo/parm.rs [hunk 2]
```diff
@@ -100,10 +91,8 @@ pub fn expand(cap: &[u8], params: &[Param], vars: &mut Variables)
     let mut stack: Vec<Param> = Vec::new();
 
     // Copy parameters into a local vector for mutability
-    let mut mparams = [
-        Number(0), Number(0), Number(0), Number(0), Number(0),
-        Number(0), Number(0), Number(0), Number(0),
-    ];
+    let mut mparams = [Number(0), Number(0), Number(0), Number(0), Number(0), Number(0),
```
Comment (Stebalien) [inline src/terminfo/parm.rs]: I *really* don't like this alignment. Is there any way to put the `Number`s on a new line?

File: src/terminfo/parm.rs [hunk 3]
```diff
@@ -414,59 +461,64 @@ impl FormatOp {
...
-                } else if d < 0 {
-                    // C doesn't take sign into account in precision calculation.
-                    format!("{:01$}", d, flags.precision + 1)
-                } else if flags.space {
-                    format!(" {:01$}", d, flags.precision)
-                } else {
-                    format!("{:01$}", d, flags.precision)
+        Number(d) => {
+            match op {
+                FormatDigit => {
```
Comment (Stebalien) [inline src/terminfo/parm.rs]: Is there any way to tell rustfmt to not put sub-matches on new lines? Putting this match one line up would save an indent.

**Comment by Stebalien** [comment]
Thanks! I like most of the changes but have a few reservations.

File: src/terminfo/parm.rs (hunk 1 above)
Comment (SingingTree) [inline src/terminfo/parm.rs]: Inconsistent in that single liners get a comma, but blocks do not?

**Comment by SingingTree** [comment]
To the best of my knowledge, none of the rustfmt configurable params cover the above cases. I'll have a mess around with it and see what can be done. However, it's possible to also manually fix the above (which will of course not help with future format runs).

File: src/terminfo/parm.rs (hunk 1 above)
Comment (Stebalien) [inline src/terminfo/parm.rs]: It's more that this needs a comma:
```rust
match x {
    Wrap(y) => match y {
         Some(z) => {}
         None => {}
     }, // <--
     _ => {}
}
```
while looking visually similar to a block.

Comment (SingingTree) [inline src/terminfo/parm.rs]: Cool. I guess that the match y is essentially a single liner to the enclosing match, thus requiring the comma. I think the issue mentioned above once sorted will give the desired style. I'll have a look at what is needed in rustfmt.

Comment (Stebalien) [inline src/terminfo/parm.rs]: This one's not *that* important; I can live with inconsistent comma placement. The one change that really bugs me is the array formatting.

**Comment by SingingTree** [comment]
I've updated this to remove the new array formatting.

**Comment by Stebalien** [comment]
@cmr, how's your error handling patch coming? Would you like me to wait on this patch?

**Comment by cmr** [comment]
@Stebalien I'll have a PR out by tomorrow, I'm almost done. I'd appreciate holding off on this, the merge conflict would have me basically rewriting the patch.

**Comment by Stebalien** [comment]
@SingingTree, Ok. All ready. When you re-run rustfmt, could you check-in the config?

**Comment by Stebalien** [comment]
Fixed in https://github.com/Stebalien/term/commit/0373197ea285b93fe37ad67ffd66cf0e5caf0eef.

**Pull request closed without merging**